### PR TITLE
Choreo Connect 1.0.0 - Helm QSG and Production deployment guide

### PR DESCRIPTION
**Pull request by renuka-fernando**
## Purpose
$subject

### CODE REVIEW

File: en/mkdocs.yml [hunk 1]
```diff
@@ -2165,6 +2166,6 @@ extra:
         - type: linkedin
           link: https://www.linkedin.com/company/wso2
     site_version: 4.0.0
-    #base_path: http://localhost:8000
-    base_path: https://apim.docs.wso2.com/en/4.0.0
+    base_path: http://localhost:8000
+#    base_path: https://apim.docs.wso2.com/en/4.0.0
```
Comment (samgnaniah) [inline en/mkdocs.yml:2170]: Undo this change.

File: en/docs/deploy-and-publish/deploy-on-gateway/deploying-apis-in-api-gateway-vs-choreo-connect.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 **API Deploying** is the process of making the API available for invocation. WSO2 API Manager comes with two gateway choices, the traditional API Gateway and the lightweight Choreo Connect.
 
-Choreo Connect is a cloud-native, open-source, and developer-centric API gateway proxy. It provides first-class support for K8s while facilitating an array of API management quality of services (QoS), such as message security rate-limiting, observability, and message mediation. It can also be configured to have multiple gateway environments which can have different sets of APIs.
+Choreo Connect is a cloud-native, open-source, and developer-centric API gateway proxy. It provides first-class support for K8s while facilitating an array of API management quality of services (QoS), such as message security, rate-limiting, observability, and message mediation. It can also be configured to have multiple gateway environments which can have different sets of APIs.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/deploying-apis-in-api-gateway-vs-choreo-connect.md:5]:
```suggestion
Choreo Connect is a cloud-native, open-source, and developer-centric API gateway proxy. It provides first-class support for K8s, while facilitating an array of API management quality of services (QoS), such as message security, rate-limiting, observability, and message mediation. It can also be configured to have multiple Gateway environments, which can have different sets of APIs.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+FROM wso2/choreo-connect-router:1.0.0
+<YOUR_DOCKER_COMMANDS>
+```
+
+```dockerfile tab='Sample'
+FROM wso2/choreo-connect-router:1.0.0
+COPY my-ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+```
+
+make sure the files you are adding or coping (with commands `ADD` and `COPY`) exists in the context path.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:280]:
```suggestion
Make sure the files you are adding or copying (with commands `ADD` and `COPY`) exist in the context path.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md [hunk 4]
```diff
@@ -83,7 +83,7 @@ Update the file inside the `Definition` folder with any API related change. The
     - API name (Title)
     - Version
     
-    The version must not be edited once deployed because in the standalone mode, the version represents the identity of the API. Therefore, editing these values will make the API behave as a new API while continuing to keep the previous version of the same API.
+    must not be edited once deployed because in the standalone mode, the version represents the identity of the API. Therefore, editing these values will make the API behave as a new API while continuing to keep the previous version of the same API.
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md:86]: Must?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
+# Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts
+
+Let's deploy an API, using WSO2 API Controller (apictl), on Choreo Connect, which running on Kubernetes as a Standalone Gateway.
+
+## Before you begin
+    
+-   Be sure you have an active [WSO2 Subscription](https://wso2.com/subscription). If you don't already have a subscription, sign up for a [WSO2 Free Trial Subscription](https://wso2.com/free-trial-subscription).
+
+    !!! Note
+        You need an active subscription to use the updated Docker images of the Micro Integrator with your Helm resources. Otherwise, you can use the community version of Docker images, which do not include product updates.
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:10]: Micro Integrator?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
+# Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts
+
+Let's deploy an API, using WSO2 API Controller (apictl), on Choreo Connect, which running on Kubernetes as a Standalone Gateway.
+
+## Before you begin
+    
+-   Be sure you have an active [WSO2 Subscription](https://wso2.com/subscription). If you don't already have a subscription, sign up for a [WSO2 Free Trial Subscription](https://wso2.com/free-trial-subscription).
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:7]: This gives the idea that ppl need to have a valid sub for using helm charts. Can we improve this as in https://github.com/wso2/kubernetes-microgateway/blob/1.0.x/helm/choreo-connect/README.md#prerequisites

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
+# Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts
+
+Let's deploy an API, using WSO2 API Controller (apictl), on Choreo Connect, which running on Kubernetes as a Standalone Gateway.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:3]:
```suggestion
Let's deploy an API using WSO2 API Controller (apictl) on Choreo Connect, which is running on Kubernetes as a Standalone Gateway.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
+# Production Deployment Guideline
+
+Given below are the common guidelines for deploying Choreo Connect in a **production environment** using [Helm deployment option](http://localhost:8000/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-deploy-overview/).
+
+## Configuring Keystore
+
+By default, Choreo Connect ships with default keystore certificates, and it is recommended to replace them with Self-Signed or CA signed certificates.
+
+### Setep 1: Generate Self-Signed certs in PEM format
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:9]:
```suggestion
### Step 1: Generate self-signed certs in PEM format
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 \
+    -subj "/C=US/ST=CA/L=Mountain View/O=WSO2, Inc./OU=Choreo Connect/CN=yourdomain.com" \
+    -extensions SAN \
+    -config <(cat /etc/ssl/openssl.cnf \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:router")) \
+    -keyout certs/router/tls.key \
+    -out certs/router/tls.crt
+```
+
+By default, SSL Hostnames for adapter and enforcer is `adapter` and `enforcer`, in case if you want to have your own SAN names in the certificates, update the following values in helm chart with the SAN names in the certificates.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:57]:
```suggestion
By default, SSL Hostnames for adapter and enforcer is `adapter` and `enforcer`, in case if you want to have your own SAN names in the certificates, update the following values in the Helm Chart with the SAN names in the certificates.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md (hunk 4 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md:86]: This is not related to the helm deployment, @suksw can u add a comment here?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 5 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md]: fixed with https://github.com/wso2/docs-apim/pull/5214/commits/d8772603ea23f01f2604f5b384e6a0bfff4b7325

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md (hunk 4 above)
Comment (suksw) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md:86]: This is a correction for a typo I made during a previous update. I requested @renuka-fernando to remove the two words because `version` is already present in the bullet points

Comment (samgnaniah) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md:86]: Shall we say something like:

"The following properties must not be edited once deployed:

- API name (Title)
- Version

The version represents the identity of the API when in standalone mode and editing these values will make the API behave as a new API, while continuing to keep the previous version of the same API."

Something like that.

Comment (suksw) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md:86]: +1. That would make it clearer. Looks like there's another typo I missed. Suggesting the exact change above without `The version` at the beginning of the paragraph. Like: 

"The following properties must not be edited once deployed:

    - API name (Title)
    - Version

A combination of these properties represents the identity of the API when in standalone mode and editing these values will make the API behave as a new API, while continuing to keep the previous version of the same API."

Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/as-a-standalone-gateway.md]: Thanks, @suksw and @samgnaniah 
fixed with https://github.com/wso2/docs-apim/pull/5214/commits/615cf57a1b30f3e47d512345bbbd470fa15f7888

**Review by suksw — APPROVED** [review]
Approving the change for `as-a-standalone-gateway.md` page.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+Choreo Connect provides the ability to generate a JWT as a test key to access the APIs. It is recommended to disable the `/testkey` endpoint in the Production environments.
+Refer the document [Generate a Test JWT]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/generate-a-test-jwt/) to read more about Test JWT.
+
+Set the following value to `false` when installing the helm chart or set it in the values.yaml file.
+
+```shell
+wso2.deployment.gatewayRuntime.enforcer.security.testTokenIssuer.enabled=false
+```
+
+## Create Custom Docker Image
```
Comment (renuka-fernando) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:263]: Make Adapter custom docker image

Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md]: Added separate sections for Standalone and Control Plane modes with https://github.com/wso2/docs-apim/pull/5214/commits/9519594eb8494b506a1b2318846d599edf1cf7f6

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
+# Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts
+
+Let's deploy an API using WSO2 API Controller (apictl) on Choreo Connect, which is running on Kubernetes as a Standalone Gateway. You can select one of the option to install Choreo Connect Helm Chart in the middle of the documentation.
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:3]: options?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+- [Option 1: Install Chart from WSO2 Helm Chart Repository](#option-1-install-chart-from-wso2-helm-chart-repository)
+- [Option 2: Install Chart from Source](#option-2-install-chart-from-source)
+
+## Before you begin
+    
+-   WSO2 product Docker images used for the Kubernetes deployment.
+
+    WSO2 product Docker images available at [DockerHub](https://hub.docker.com/u/wso2/) package General Availability (GA)
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:12]: WSO2 product Docker images available at [DockerHub](https://hub.docker.com/u/wso2/) (General Availability (GA)
    versions of WSO2 products with no [WSO2 Updates](https://wso2.com/updates).)

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+    ```bash
+    <EXTERNAL-IP>   adapter.wso2.com
+    <EXTERNAL-IP>   gw.wso2.com
+    ```
+
+3.  Execute the following command to invoke health check services:
+
+    ```bash
+    curl -X GET "https://gw.wso2.com/health"
+    ```
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:188]: Let's remove this step 3.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-on-kubernetes-with-apim-as-control-plane-helm-artifacts.md [hunk 14]
```diff
@@ -0,0 +1,293 @@
...
+    ```bash
+    <EXTERNAL-IP>   am.wso2.com
+    <EXTERNAL-IP>   gw.wso2.com
+    ```
+
+3.  Execute the following command to invoke health check services:
+
+    ```bash
+    curl -X GET "https://gw.wso2.com/health"
+    ```
```
Comment (pubudu538) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-on-kubernetes-with-apim-as-control-plane-helm-artifacts.md:286]: Let's remove this.

Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-on-kubernetes-with-apim-as-control-plane-helm-artifacts.md]: fixed with https://github.com/wso2/docs-apim/pull/5214/commits/56e8d06f0eefdc6ceb6870305d7af0214650c154

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 13 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md]: fixed with https://github.com/wso2/docs-apim/pull/5214/commits/56e8d06f0eefdc6ceb6870305d7af0214650c154

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 12 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md]: fixed with https://github.com/wso2/docs-apim/pull/5214/commits/56e8d06f0eefdc6ceb6870305d7af0214650c154

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md (hunk 11 above)
Comment (renuka-fernando) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md]: fixed with https://github.com/wso2/docs-apim/pull/5214/commits/56e8d06f0eefdc6ceb6870305d7af0214650c154

**Review by pubudu538 — APPROVED** [review]

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+
+## Before you begin
+    
+-   WSO2 product Docker images used for the Kubernetes deployment.
+
+    WSO2 product Docker images available at [DockerHub](https://hub.docker.com/u/wso2/) (General Availability (GA)
+    versions of WSO2 products with no [WSO2 Updates](https://wso2.com/updates).
+
+    For a production grade deployment of the desired WSO2 product-version, it is highly recommended to use the relevant
+    Docker image which packages WSO2 Updates, available at [WSO2 Private Docker Registry](https://docker.wso2.com/). In order
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:16]:
```suggestion
    Docker image that packages WSO2 Updates, available at [WSO2 Private Docker Registry](https://docker.wso2.com/). In order
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+    ``` tab='Format'
+    helm install <RELEASE_NAME> wso2/choreo-connect --version 1.0.0-1 --namespace <NAMESPACE> --create-namespace
+    ```
+
+    ``` tab='Sample'
+    helm install my-release wso2/choreo-connect --version 1.0.0-1 --namespace cc --create-namespace
+    ```
+
+The above steps will deploy the Choreo Connect using WSO2 product Docker images available at DockerHub.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts.md:64]:
```suggestion
The above steps will deploy Choreo Connect using WSO2 product Docker images available at DockerHub.
```

**Review by samgnaniah — APPROVED** [review]
